android/src/main/kotlin/com/jhomlala/better_player/BetterPlayerPlugin.kt

```diff
@@ -511,15 +547,17 @@ class BetterPlayerPlugin : FlutterPlugin, ActivityAware, MethodCallHandler {
             val textureId = getTextureId(betterPlayer)
             if (textureId != null) {
                 val dataSource = dataSources[textureId]
-                //Don't setup notification for the same source.
-                if (textureId == currentNotificationTextureId && currentNotificationDataSource != null && dataSource != null && currentNotificationDataSource === dataSource) {
-                    return
-                }
-                currentNotificationDataSource = dataSource
-                currentNotificationTextureId = textureId
-                removeOtherNotificationListeners()
                 val showNotification = getParameter(dataSource, SHOW_NOTIFICATION_PARAMETER, false)
                 if (showNotification) {
```

Comment:
Move code for checking showNotification flag. 
Since even if same `textureId` and `datasource` , we need to `betterPlayer.reactivateMediaSessionIfNeeded()` now.